### PR TITLE
Strict scoping for global services

### PR DESCRIPTION
**Pull request by alllex**
This PR requires all services registered in Global service registries to be explicitly annotated with `@ServiceScope(Global)`.

This is an internal refactoring and should not have any user-facing changes.

Since it touches a lot of files, the change is split into two commits to make it easier to review.

1. First, large but simple change. We annotate all Global services with the appropriate annotation and also explicitly state which services some implementations declare.
2. Then enable the validation. We enforce the services to be annotated (with some exceptions), ensuring that no services are left behind.

The services for which the validation is suppressed are not currently supported by validation logic. These are either services that involve generics, or services that require further consideration from the owning team on how to expose them more cleanly.

### CODE REVIEW

**Comment by alllex** [comment]
@bot-gradle test

**Comment by alllex** [comment]
@bot-gradle test qfl

**Comment by alllex** [comment]
@bot-gradle test without pts

**Review by bamboo — APPROVED** [review]
👍

File: platforms/core-runtime/base-services/src/main/java/org/gradle/internal/service/ServiceScopeValidatorWorkarounds.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        "org.gradle.nativeplatform.platform.internal.NativePlatforms",
+        "org.gradle.nativeplatform.internal.NativePlatformResolver",
+        "org.gradle.nativeplatform.internal.DefaultTargetMachineFactory",
+        "org.gradle.nativeplatform.test.internal.NativeTestSuiteBinaryRenderer",
+
+        "org.gradle.internal.scripts.DefaultScriptFileResolverListeners"
+    ));
+
+    public static boolean shouldSuppressValidation(Class<?> serviceType) {
+        return SUPPRESSED_VALIDATION_CLASSES.contains(serviceType.getName());
```
Comment (tresat) [inline platforms/core-runtime/base-services/src/main/java/org/gradle/internal/service/ServiceScopeValidatorWorkarounds.java:59]: 💭 Should this also check if a `serviceType` extends one of these suppressed types?

**Review by tresat — APPROVED** [review]

**Review by adammurdoch — APPROVED** [review]

File: platforms/core-runtime/base-services/src/main/java/org/gradle/internal/service/ServiceScopeValidatorWorkarounds.java (hunk 1 above)
Comment (alllex) [reply platforms/core-runtime/base-services/src/main/java/org/gradle/internal/service/ServiceScopeValidatorWorkarounds.java:61]: I think it's better to be explicit here as we want to reduce and remove the suppressed list soon.

**Pull request merged**